kernel/Utilities/SparsityEstimator.cpp

```diff
@@ -15,310 +18,416 @@ SparsityEstimator::SparsityEstimator(const GlobalIndexing& rowindexing,
+            // Check for a distributed mesh
+            if (face->getFaceType() == Geometry::FaceType::SUBDOMAIN_BOUNDARY ||
+                face->getFaceType() ==
+                    Geometry::FaceType::PERIODIC_SUBDOMAIN_BC) {
+                distributedMesh = true;
             }
-            workspace.clear();
-            for (const Base::Element* element : node->getElements()) {
-                // DoFs from the supporting element
-                addElementDoFs(element, workspace);
-                // Face matrix coupling with the element on the other side of
-                // the face
-                if (!includeFaceCoupling) continue;
-                // See remark for face based DoFs
-                logger.assert_always(
-                    element->isOwnedByCurrentProcessor(),
-                    "Face coupling with conforming basis functions on "
-                    "distributed meshes is not supported");
-                for (const Base::Face* face : element->getFacesList()) {
-                    if (!face->isInternal()) continue;
-                    const Base::Element* otherElement =
-                        face->getPtrOtherElement(element);
-                    addElementDoFs(otherElement, workspace);
+            // Actual face matrix
+            rowDoFs.clear();
+            columnDoFs.clear();
+            rowDoFs.addLocalFaceMatrixDoFs(face);
+            columnDoFs.addLocalFaceMatrixDoFs(face);
+            // Add the DoFs
+            for (std::size_t ri = 0;
+                 ri < rowIndexing_.getNumberOfIncludedUnknowns(); ++ri) {
+                std::size_t rowUnknown = rowIndexing_.getIncludedUnknowns()[ri];
+                const Detail::DoFIntervals& rowIntervals =
+                    rowDoFs.getOwnedIntervals(rowUnknown);
+
+                for (std::size_t ci = 0;
+                     ci < columnIndexing_.getNumberOfIncludedUnknowns(); ++ci) {
+                    std::size_t colUnknown =
+                        columnIndexing_.getIncludedUnknowns()[ci];
+                    if (!faceCoupling(ri, ci)) {
+                        continue;
+                    }
+
+                    const Detail::DoFIntervals& ownedColIntervals =
+                        columnDoFs.getOwnedIntervals(colUnknown);
+                    const Detail::DoFIntervals& nonOwnedColIntervals =
+                        columnDoFs.getNonOwnedIntervals(colUnknown);
+
+                    for (Detail::DoFInterval rowInterval : rowIntervals) {
+                        for (Detail::DoFInterval colInterval :
+                             ownedColIntervals) {
+                            owned[rowInterval].emplace(colInterval);
+                        }
+                        for (Detail::DoFInterval colInterval :
+                             nonOwnedColIntervals) {
+                            nonOwned[rowInterval].emplace(colInterval);
+                        }
```

Comment:
an insert call should be enough here I think

Reply:
Replaced.